src/unused_finder/mod.rs

```diff
@@ -110,107 +124,91 @@ pub fn find_unused_items(
-                                    match imported {
-                                        node_visitor::ImportedItem::ExecutionOnly
-                                        | node_visitor::ImportedItem::Namespace => {
-                                            // Even if exported elements are not imported specifically, side effects take place
-                                            // In case of namespace import (import * as foo from 'foo'), an exhaustive search on which items of `foo` are used.
-                                            // For now, assume that all items are used and by clearing the map we mark all items as used
-                                            // TODO add node visitor that does search for specific items used.
-                                            origin_file_exported_items.clear();
-                                            unused_files.remove(imported_path);
-                                        }
-                                        _ => {
-                                            unused_files.remove(imported_path);
-                                            let i = ExportKind::from(imported);
-                                            origin_file_exported_items.retain(|exp_item| {
-                                                exp_item.metadata.export_type != i
-                                            });
-                                            // origin_file_exported_items.remove(&i);
-                                        }
-                                    }
-                                }
-                                ResolvedItem::Exported(_) => {}
-                            }
-                        });
-                    }
-                });
-        });
-    let mut unused_items_len = 0;
-
-    // Print the unused items for each file, sort them by file first
-    let mut path_unused_items: Vec<_> = path_unused_items_map.into_iter().collect();
-    path_unused_items.sort_by(|x, y| x.0.cmp(&y.0));
-    path_unused_items.iter_mut().for_each(|(k, items)| {
-        items.retain(|i| !i.metadata.allow_unused);
-        if items.len() > 0 {
-            if !unused_files.contains(k) {
-                println!("From file {} found {:?} unused items", k, items);
+    let entry_files: Vec<String> = flattened_walk_file_data
+        .iter_mut()
+        .filter_map(|file| {
+            if entry_packages.contains(&file.package_name) {
+                return Some(file.source_file_path.clone());
             }
+            None
+        })
+        .collect();
+    let mut new_entries = entry_files;
+    let max_iters = 10_000_000;
+    let mut current_iter = 0;
+    loop {
+        current_iter += 1;
+        new_entries = graph.bfs(new_entries);
+        if current_iter > max_iters || new_entries.is_empty() {
+            break;
         }
-        unused_items_len += items.len();
-    });
+    }
 
     let results: Vec<String> = Vec::new();
```

Comment:
This should be populated with unused_files, right now it's just returned as an empty vector?